scvi/external/cellassign/_module.py

```diff
@@ -0,0 +1,188 @@
+import numpy as np
+import torch
+
+# import pdb
+import torch.nn.functional as F
+
+from scvi import _CONSTANTS
+from scvi.compose import BaseModuleClass, LossRecorder, auto_move_data
+from scvi.distributions import NegativeBinomial
+from torch.distributions import Dirichlet, Normal
+
+
+class CellAssignModule(BaseModuleClass):
+    """
+    Model for CellAssign.
+
+    Parameters
+    ----------
+    n_genes
+        Number of input genes
+    n_labels
+        Number of input cell types
+    rho
+        Binary matrix of cell type markers
+    **model_kwargs
+        Additional kwargs
+    """
+
+    def __init__(
+        self,
+        n_genes: int,
+        n_labels: int,
+        rho: torch.Tensor,
+        **model_kwargs,
+    ):
+        super().__init__()
+        self.n_genes = n_genes
+        self.n_labels = n_labels
+        self.rho = rho
+
+        self.register_buffer("cell_type_markers", rho)
+
+        # perform all other initialization
+        self.LOWER_BOUND = 1e-10
+        self.THETA_LOWER_BOUND = 1e-20
```

Comment:
Not sure these need to be class attributes? They could just be near the imports?